### PR TITLE
Fixed an issue for periodic boundary

### LINKED ISSUE #7946: Make DisplacedSystem and NonlinearSystem consistent for periodic boundary condition
`DisplacedSystem` has a shorter ghosted vector than `NonlinearSystem` if its `DofMap` is not updated accordingly when creating a periodic boundary condition.

### PR DESCRIPTION
**Pull request by fdkong**
Make `DisplacedSystem` and `NonlinearSystem` consistent for periodic boundary condition.

`DisplacedSystem` has a shorter ghosted vector than `NonlinearSystem` if its `DofMap` is not updated accordingly when creating a periodic boundary condition.

This commit fixed this issue by updating `DofMap` accordingly.

Closes #7946

### CODE REVIEW

File: modules/combined/tests/periodic_boundary/thinfilm_periodic_boundary.i [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+  type = Transient
+    [./TimeStepper]
+    type = IterationAdaptiveDT
+    dt = 0.8
+    #iteration_window = 3
+    optimal_iterations = 6 #should be 5 probably
+    growth_factor = 1.4
+    linear_iteration_ratio = 1000
+    cutback_factor =  0.8
+[../]
```
Comment (dschwen) [inline modules/combined/tests/periodic_boundary/thinfilm_periodic_boundary.i]: could this input file get cleaned up a bit? maybe?

File: framework/src/actions/AddPeriodicBCAction.C [hunk 2]
```diff
@@ -68,6 +69,8 @@ AddPeriodicBCAction::setPeriodicVars(PeriodicBoundaryBase & p, const std::vector
 bool
 AddPeriodicBCAction::autoTranslationBoundaries()
 {
+  MooseSharedPointer<DisplacedProblem> displaced_problem = _problem->getDisplacedProblem();
```
Comment (dschwen) [inline framework/src/actions/AddPeriodicBCAction.C]: those seem like good places for `auto`

File: framework/src/actions/AddPeriodicBCAction.C [hunk 3]
```diff
@@ -152,6 +158,8 @@ AddPeriodicBCAction::act()
     _problem->addGhostedBoundary(p.pairedboundary);
 
     nl.dofMap().add_periodic_boundary(p);
+    if (displaced_problem != NULL)
```
Comment (fdkong) [inline framework/src/actions/AddPeriodicBCAction.C]: Like` if(displaced_problem)`?  What I am using here is the same as anywhere else. Honestly, I paste it from `FEProblem` :)

**Comment by dschwen** [comment]
Hmmm, the test you added exodiffs.

**Comment by mangerij** [comment]
There probably is a nullspace since if you rotate the mesh 90 degrees you
could have the same solution.

**Comment by fdkong** [comment]
> There probably is a nullspace since if you rotate the mesh 90 degrees you could have the same solution.

@mangerij  Did you remove the null space or not? Then you could have an unique solution with the minimal norm.

**Comment by mangerij** [comment]
Hmm, OK, I've updated the input film to remove the null space. How do I check out your branch and commit to this PR?

**Comment by fdkong** [comment]
> Hmm, OK, I've updated the input film to remove the null space. How do I check out your branch and commit to this PR?

git checkout -b fdkong-fix_different_vector_length devel
git pull https://github.com/fdkong/moose.git fix_different_vector_length

**Comment by mangerij** [comment]
Alright all fixed.

Now just git push and it goes to your repo right?

**Comment by mangerij** [comment]
That input file was mine :P

There is nothing to say you need iterative adaptiveDT. You can dump that
stuff. Also, those petsc options don't need to be there.

**Comment by fdkong** [comment]
> Alright all fixed. Now just git push and it goes to your repo right?

Should go to your repo, and use your own PR

**Comment by fdkong** [comment]
Moved to #7955

**Pull request closed without merging**